extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php

```diff
@@ -34,11 +49,72 @@ static public function onImageServingCollectImages( &$imageNamesArray, $articleT
 	 * @param $parser
 	 * @param $frame
 	 */
-	static public function onParserTagHooksBeforeInvoke( $name, $marker, $content, $attributes, $parser, $frame ) {
+	public static function onParserTagHooksBeforeInvoke( $name, $marker, $content, $attributes, $parser, $frame ) {
 		if ( $name === self::PARSER_TAG_GALLERY ) {
 			\Wikia\PortableInfobox\Helpers\PortableInfoboxDataBag::getInstance()->setGallery( $marker, $content );
 		}
 
 		return true;
 	}
+
+	/**
+	 * renders portable infobox builder instead of edit page while creating new template
+	 *
+	 * @param EditPageLayoutController $editPageContext
+	 * @return bool
+	 */
+	public static function onEditPageLayoutExecute( $editPageContext ) {
+		$context = $editPageContext->getContext();
+		$webRequest = $context->getRequest();
+		$title = $context->getTitle();
+
+		if ( self::isEditingNewTemplate( $webRequest, $title ) && $webRequest->getVal( self::INFOBOX_BUILDER_QS_PARAM, false
```

Comment:
Maybe it's worth to extract the second condition into self::isInfoboxBuilderRequested? It'll make it more readable here and in the neagtion in L96